src/include/sof/lib/alloc.h

```diff
@@ -106,10 +106,11 @@ struct mm {
 /* heap allocation and free */
 void *_malloc(int zone, uint32_t caps, size_t bytes);
 void *_zalloc(int zone, uint32_t caps, size_t bytes);
-void *_balloc(int zone, uint32_t caps, size_t bytes, uint32_t alignment);
+void *_balloc(int zone, uint32_t caps, size_t bytes, uint32_t alignment,
+	      bool can_fail);
 void *_realloc(void *ptr, int zone, uint32_t caps, size_t bytes);
```

Comment:
Instead of adding a new parameter I think we can extend the caps field to also include a flag named SOF_MEM_NOWARN (similar with kerne's GFP_NOWARN).

When this flags is set it has the same meaning as your can_fail parameter.

Also, it might worth renaming caps parameter to flags.

Reply:
I was thinking about that too. However I am afraid it may confuse people more than extra single argument. Anyway it is worth discussing - if more people like this idea too I am more than happy to go with this direction.

Reply:
caps is for capabilities, not error reporting so it would be confusing, I'd go for another function

Reply:
@jajanusz will rename parameter from caps to flags and so that it can support more features. Have a look at GFP_  macros in Linux kernel (GFP_NOWARN).

Reply:
@dbaluta If you change it to `flags` then it's fine for me